app/models/manageiq/providers/google/cloud_manager/event_catcher/runner.rb

```diff
@@ -0,0 +1,74 @@
+class ManageIQ::Providers::Google::CloudManager::EventCatcher::Runner <
+  ManageIQ::Providers::BaseManager::EventCatcher::Runner
+
+  # Request the event monitor to stop running.
+  def stop_event_monitor
+    stop_running
+  end
+
+  # Start monitoring for events. This method blocks forever until
+  # #stop_event_monitor is called.
+  def monitor_events
+    _log.info "#{log_prefix} Monitoring for events"
+
+    start_running
+    event_monitor_running
+    while running?
+      heartbeat
+      events = event_producer.poll
+      heartbeat
+
+      _log.info "#{log_prefix} Received #{events.size} events" unless events.empty?
+      events.each do |event|
+        _log.debug "#{log_prefix} Event: #{event}"
+        @queue.enq event if event_requires_refresh?(event)
```

Comment:
Is the word `refresh` here implying an EmsRefresh or does it mean something else?

As @blomquisg said, we typically collect *all* events, so they can be displayed on a timeline, with rare exceptions in the blacklisted events table, so the check should not be completely necessary (unless, perhaps you wanted to keep the bit about an incomplete event).